### PR TITLE
Allow zero-offset in OffsetView

### PR DESCRIPTION
**Pull request by nliber**
Working on test cases for this...

### CODE REVIEW

**Comment by dalg24** [comment]
@nliber please test rank-2 and rank-3 offset views as well

File: containers/unit_tests/TestOffsetView.hpp [hunk 1]
```diff
@@ -461,33 +461,118 @@ void test_offsetview_subview(unsigned int size) {
...
-  for (int i = 0; i != v.extent_int(0); ++i) v(i) = static_cast<SCALAR>(i);
-
-  // Test offsets -3..3, including 0
-  for (int j = -3; j != 3; ++j) {
-    offset_view_type ov(v, {j});
-    ASSERT_EQ(ov(3), static_cast<SCALAR>(3 - j))
-        << "offset " << j << " is broken.";
-  }
+KOKKOS_INLINE_FUNCTION int element(std::initializer_list<int> il)
+{ return std::accumulate(il.begin(), il.end(), 0, [](int l, int r) { return l * 10 + r; }); }
```
[the remark this answers is not in the document]
Comment (mhoemmen) [reply containers/unit_tests/TestOffsetView.hpp]: Does `std::initializer_list` work on device?

[the remark this answers is not in the document]
Comment (dalg24) [reply containers/unit_tests/TestOffsetView.hpp]: I think it is supported since CUDA 7

[the remark this answers is not in the document]
Comment (nliber) [reply containers/unit_tests/TestOffsetView.hpp]: I don't need to use initializer_list.  Should I change it?

[the remark this answers is not in the document]
Comment (nliber) [reply containers/unit_tests/TestOffsetView.hpp]: Not sure where <numeric> is coming from, as I didn't add any headers.

**Comment by dalg24** [comment]
I think this is ready for review.

File: containers/unit_tests/TestOffsetView.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (dalg24) [reply containers/unit_tests/TestOffsetView.hpp]: No

**Comment by dalg24** [comment]
Reference #2246 (somehow review comment didn't do it)

**Pull request merged**